packages/flutter_riverpod/lib/src/framework.dart

```diff
@@ -166,14 +169,20 @@ class ProviderScope extends StatefulWidget {
   /// Informations on how to override a provider/family.
   final List<Override> overrides;
 
+  /// {@macro flutter.widgets.widgetsApp.restorationScopeId}
+  final String? restorationId;
+
+  /// Registers global [RestorableProvider]s for restoration.
+  final List<RestorableRestorationId> restorables;
```

Comment:
This shouldn't be needed. It goes backward to how Riverpod works


Restorables should automatically be registered/unregistered when providers are mounted/unmounted